### PR TITLE
JAMES-2973 Add a test ensuring JMAP ability to access to 'Outbox'

### PR DESCRIPTION
**Pull request by blackheaven**

### CODE REVIEW

**Comment by chibenwa** [comment]
Giving it a second thought, I would bet qa issue is due to ElasticSearch index delay...

**Comment by Arsnael** [comment]
```
[ERROR] Errors: 
[cf485430b85e63ec768e84364896dec68310b099] [ERROR]   WebAdminServerIntegrationTest.setUp:96 » Retryable Error encountered, scheduli...
[cf485430b85e63ec768e84364896dec68310b099] [ERROR]   WebAdminServerIntegrationTest.setUp:96 » Retryable Error encountered, scheduli...
[cf485430b85e63ec768e84364896dec68310b099] [ERROR]   WebAdminServerIntegrationTest.setUp:96 » Retryable Error encountered, scheduli...
[cf485430b85e63ec768e84364896dec68310b099] [ERROR]   WebAdminServerIntegrationTest.setUp:96 » Retryable Error encountered, scheduli...
[cf485430b85e63ec768e84364896dec68310b099] [ERROR]   WebAdminServerIntegrationTest.setUp:96 » Retryable Error encountered, scheduli...
[cf485430b85e63ec768e84364896dec68310b099] [ERROR]   WebAdminServerIntegrationTest.setUp:96 » Retryable Error encountered, scheduli...
[cf485430b85e63ec768e84364896dec68310b099] [ERROR]   WebAdminServerIntegrationTest.setUp:96 » Configuration Unable to init mailet R...
```

**Comment by blackheaven** [comment]
test this please

File: server/protocols/jmap-draft-integration-testing/jmap-draft-integration-testing-common/src/test/java/org/apache/james/jmap/draft/methods/integration/GetMessageListMethodTest.java [hunk 1]
```diff
@@ -2319,6 +2320,24 @@ public void getMessageListShouldReturnTwoMessagesWhenCopiedAtOnceViaIMAP() throw
...
 
+    @Test
+    public void sendingAMessageThroughAnotherProtocolShouldBePresentInOutboxViaJmap() throws Exception {
+        MailboxId outboxMailboxId = mailboxProbe.createMailbox(MailboxConstants.USER_NAMESPACE, ALICE, "Outbox");
+        String messageBody = "We're all mad here.";
+        mailboxProbe.appendMessage(
+            ALICE,
+            MailboxPath.forUser(ALICE, "Outbox"),
+            new ByteArrayInputStream(("Subject: test\r\n\r\n" + messageBody).getBytes(StandardCharsets.UTF_8)),
+            new Date(), false, new Flags());
```
Comment (chibenwa) [inline server/protocols/jmap-draft-integration-testing/jmap-draft-integration-testing-common/src/test/java/org/apache/james/jmap/draft/methods/integration/GetMessageListMethodTest.java]: Refactoring idea: adding a fluent helper class taking a damn `AppendCommand` as a unique parameter...

**Comment by chibenwa** [comment]
Build aborted.

test this please

**Comment by chibenwa** [comment]
```
[cf485430b85e63ec768e84364896dec68310b099] [ERROR] Failed to execute goal on project apache-james-backends-jpa: Could not resolve dependencies for project org.apache.james:apache-james-backends-jpa:jar:3.5.0-SNAPSHOT: Failed to collect dependencies at com.google.guava:guava:jar:25.1-jre -> org.codehaus.mojo:animal-sniffer-annotations:jar:1.14: Failed to read artifact descriptor for org.codehaus.mojo:animal-sniffer-annotations:jar:1.14: Could not transfer artifact org.codehaus.mojo:animal-sniffer-annotations:pom:1.14 from/to central (https://repo.maven.apache.org/maven2): /root/.m2/repository/org/codehaus/mojo/animal-sniffer-annotations/1.14/animal-sniffer-annotations-1.14.pom.part (No such file or directory) -> [Help 1]
```

test this please

**Comment by chibenwa** [comment]
`CassandraLdapJamesServerTest.mailsShouldBeWellReceivedBeforeFirstUserConnectionWithLdap:84 » ConditionTimeout`

test this please

File: server/protocols/jmap-draft-integration-testing/jmap-draft-integration-testing-common/src/test/java/org/apache/james/jmap/draft/methods/integration/GetMessageListMethodTest.java [hunk 2]
```diff
@@ -2319,6 +2320,24 @@ public void getMessageListShouldReturnTwoMessagesWhenCopiedAtOnceViaIMAP() throw
             .until(() -> twoMessagesFoundInMailbox(otherMailboxId));
     }
 
+    @Test
+    public void sendingAMessageThroughAnotherProtocolShouldBePresentInOutboxViaJmap() throws Exception {
+        MailboxId outboxMailboxId = mailboxProbe.createMailbox(MailboxConstants.USER_NAMESPACE, ALICE, "Outbox");
+        String messageBody = "We're all mad here.";
+        mailboxProbe.appendMessage(
```
Comment (mbaechler) [inline server/protocols/jmap-draft-integration-testing/jmap-draft-integration-testing-common/src/test/java/org/apache/james/jmap/draft/methods/integration/GetMessageListMethodTest.java]: it's not a message sending, the test name is no consistent with the fact you put it through internal APIs

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**